### PR TITLE
MacOS Build Support

### PR DESCRIPTION
**Pull request by zoul0813**
* add `detected_OS` in Makefile
* add Darwin support for `stat`

Requires `brew install make` (ie; GNU Make v4+, MacOS ships with GNU Make 3)
Use `gmake` in place of `make` on MacOS

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Set the binaries
 SHELL := /bin/bash
+
+ifeq '$(findstring ;,$(PATH))' ';'
```
Comment (Zeal8bit) [inline Makefile:4]: To detect the OS, you should be able to use `uname` command. In my case on my Linux, `uname` returns `Linux`, on Mac it should return `Darwin`, on Windows it should return `WindowsNT`.

Another option is to use the macro `OSTYPE`.

File: target/zeal8bit/unit.mk [hunk 2]
```diff
@@ -37,8 +45,9 @@ endif
 # In our case, compile the programs that will be part of ROMDISK and create it.
 # After creation, get its size, thanks to `stat` command, and store it in a generated header file
 # named `romdisk_info_h.asm`
-PRECMD := (cd $(ZOS_PATH)/romdisk && make) && \
-          SIZE=$$(stat -c %s $(ZOS_PATH)/romdisk/disk.img) && \
+PRECMD := echo "Detected $(detected_OS) - $(STAT_BYTES)" && \
+					(cd $(ZOS_PATH)/romdisk && make) && \
```
Comment (Zeal8bit) [inline target/zeal8bit/unit.mk:49]: Indentation issue, may the spaces were replaced by tabs?

**Review by Zeal8bit — COMMENTED** [review]
Thanks a lot for your PR, this will help Mac users!

I confirm your changes work on my Ubuntu computer.

File: Makefile (hunk 1 above)
Comment (zoul0813) [reply Makefile:4]: `uname` is not generally available on Windows, and `OSTYPE` is an environment variable commonly available on shells, but not guaranteed.

The else condition does use `uname` for Linux and Darwin, but it attempts to detect Windows first as *nix systems use `:` for path separators, and windows uses `;` this trick allows it to quickly identify Windows.

In the event the user is using something like CYGWIN or MINGW ... these are also identified.

It's probably a bit much, TBH ... as my goal was to just get things working on a Mac, but laying the foundation for other possible platforms at this stage seemed like a good idea.

File: kernel_headers/examples/sdcc/Makefile [hunk 3]
```diff
@@ -40,6 +60,7 @@ SRCS_REL=$(patsubst %.c,%.rel,$(SRCS_OUT_DIR))
 
 all: clean $(OUTPUT_DIR) $(OUTPUT_DIR)/$(BIN_HEX) $(OUTPUT_DIR)/$(BIN)
 	@bash -c 'echo -e "\x1b[32;1mSuccess, binary generated: $(OUTPUT_DIR)/$(BIN)\x1b[0m"'
+	@echo "uartrcv $$($(STAT_BYTES) $(OUTPUT_DIR)/$(BIN)) $(BIN)"
```
Comment (Zeal8bit) [inline kernel_headers/examples/sdcc/Makefile:63]: debug leftover?

**Pull request merged**